pandemania/src/components/graphs/AreaDateTimeCountryData/AreaDateTimeCountryData.jsx

```diff
@@ -0,0 +1,161 @@
+/* eslint-disable no-case-declarations */
+import React, { useEffect, useState } from 'react';
+import propTypes from 'prop-types';
+import Chart from 'react-apexcharts';
+import ApexCharts from 'apexcharts';
+import Button from '@material-ui/core/Button';
+import ButtonGroup from '@material-ui/core/ButtonGroup';
+import dataForCountry from '../../../api/FetchDataForCountry';
+import './AreaDateTimeCountryData.css';
+
+const id = 'area-datetime';
+
+const initialState = {
+	series: [
+		{
+			data: [[new Date().getTime(), 0]],
+		},
+	],
+	options: {
+		chart: {
+			id,
+			zoom: {
+				autoScaleYaxis: true,
+			},
+		},
+		theme: {
+			mode: 'dark',
+		},
+		dataLabels: {
+			enabled: false,
+		},
+		xaxis: {
+			type: 'datetime',
+		},
+		tooltip: {
+			x: {
+				format: 'dd MMM yyyy',
+			},
+		},
+	},
+	selection: 'one_year',
+};
+
+const AreaDateTimeCountryData = ({ countryName }) => {
+	const [chartData, setChartData] = useState(initialState);
+
+	const updateData = (timeline) => {
+		setChartData({
+			...chartData,
+			selection: timeline,
+		});
+
+		const dates = chartData.series[0].data;
+		const length = dates.length - 1;
+		let daysToGoBack;
+
+		switch (timeline) {
+			case 'one_month':
+				daysToGoBack = 31;
+				ApexCharts.exec(id, 'zoomX', dates[length - daysToGoBack][0], dates[length][0]);
+				break;
+			case 'six_months':
+				daysToGoBack = 6 * 31;
+				ApexCharts.exec(id, 'zoomX', dates[length - daysToGoBack][0], dates[dates.length - 1][0]);
+				break;
+			case 'one_year':
+				daysToGoBack = 365;
+				ApexCharts.exec(id, 'zoomX', dates[length - daysToGoBack][0], dates[length][0]);
+				break;
+			case 'ytd':
+				const firstDay = new Date(new Date().getFullYear(), 0, 1);
+				ApexCharts.exec(id, 'zoomX', firstDay.getTime(), dates[length][0]);
+				break;
+			case 'all':
+				ApexCharts.exec(id, 'zoomX', dates[0][0], dates[length][0]);
+				break;
+			default:
+		}
+	};
+
+	const fetchData = async () => {
+		const response = await dataForCountry(countryName);
+
+		const series = [
+			{
+				name: 'Confirmed cases',
+				data: response.map((c) => [c.date, c.confirmed]),
+			},
+			{
+				name: 'Recovered cases',
+				data: response.map((c) => [c.date, c.recovered]),
+			},
+		];
+
+		setChartData({
+			...chartData,
+			series,
+		});
+	};
+
+	useEffect(() => {
+		fetchData();
+	}, []);
+
+	return (
+		<>
+			<ButtonGroup className="toolbar">
+				<Button
```

Comment:
you can create a configuration object like this one:
```
[
  {
    value: 'one_month',
    label: '1M'
  },
  ...
]
```
and loop through it to create `<Button>` element. That way you avoid markup hardcoding:
https://alexkondov.com/tao-of-react/#dont-hardcode-markup

Reply:
Good idea 👍.
Done!